demos/common/cpp/models/include/models/model_base.h

```diff
@@ -44,8 +45,12 @@ class ModelBase {
     }
 
 protected:
+    InferenceEngine::CNNNetwork prepareNetwork(InferenceEngine::Core& core);
+    virtual void prepareInputsOutputs(InferenceEngine::CNNNetwork& cnnNetwork) = 0;
+
     std::vector<std::string> inputsNames;
     std::vector<std::string> outputsNames;
-    InferenceEngine::ExecutableNetwork* execNetwork;
+    InferenceEngine::ExecutableNetwork execNetwork;
     std::string modelFileName;
+    CnnConfig cnnConfig = {};
```

Comment:
Do you really need `cnnConfig` as attribute?

Reply:
It contains useful information that may be used inside model functions. It will be used in following code updates, but as long as it is belongs to refactoring I've left it here